bindings/rust/s2n-tls/src/config.rs

```diff
@@ -291,6 +298,79 @@ impl Builder {
         Ok(self)
     }
 
+    /// Corresponds to [s2n_config_add_cert_chain_and_key_to_store].
+    pub fn load_chain(&mut self, chain: CertificateChain<'static>) -> Result<&mut Self, Error> {
+        // Out of an abudance of caution, we hold a reference to the CertificateChain
+        // regardless of whether add_to_store fails or succeeds. We have limited
+        // visibility into the failure modes, so this behavior ensures that _if_
+        // the C library held the reference despite the failure, it would continue
+        // to be valid memory.
+        let result = unsafe {
+            s2n_config_add_cert_chain_and_key_to_store(
+                self.as_mut_ptr(),
+                // SAFETY: audit of add_to_store shows that the certificate chain
+                // is not mutated. https://github.com/aws/s2n-tls/issues/4140
+                chain.as_ptr() as *mut _,
+            )
+            .into_result()
+        };
+        self.context_mut().application_owned_certs.push(chain);
+        result?;
+
+        Ok(self)
+    }
+
+    /// Corresponds to [s2n_config_set_cert_chain_and_key_defaults].
+    pub fn set_default_chains<T: IntoIterator<Item = CertificateChain<'static>>>(
+        &mut self,
+        chains: T,
+    ) -> Result<&mut Self, Error> {
+        // Must be greater than or equal to S2N_CERT_TYPE_COUNT in s2n_certificate.h,
+        // and small enough that stack allocating an array of this length is safe.
+        const CHAINS_MAX_COUNT: usize = 5;
+
```

Comment:
Bleh, I think I was wrong about setting this to anything other than 3. We have the test to keep it up to date anyway, might as well make it match exactly. Otherwise you'll get a rust error for 6 chains but a C error for 4 chains, which is kind of wonky.

The new name is much better though.